### ISSUE TITLE
SCAN2 for "clonal" sSNVs?

### ISSUE THREAD

**Issue by proukakis** [parklab/SCAN2#9]
hello, i appreciate the logic for not reporting any mutations with supporting reads in bulk WGS, but this probably is redundant if there is matched high coverage bulk genome eg 100x WGS- even low clonal mutations might have supporting reads. Could there be an option to report separately the single cell putative sSNVs with a low number of matching bulk reads, (to be set manually)? Thanks

**Comment by jluquette** [comment]
Hi @proukakis,

We are indeed working on allowing the user to relax the bulk filter (i.e., currently 0 mutation supporting reads in bulk are tolerated; options to increase this by count or by VAF in bulk are being added).

The tricky part isn't changing those bulk filters, but rather re-assessing the tool's performance. We don't know how many false positives (either artifacts that bulks help to remove or germline mutations) will be included if the bulk filter is relaxed.

**Comment by alextidd** [comment]
Hi @jluquette,
Is there any update on this feature? I am trying to run SCAN2 to call somatic mutations in immune cells, so there is a chance of a few reads from mobile immune clones infiltrating the bulk sample. Being able to relax this filter would be very useful.
Thanks!

**Comment by jluquette** [comment]
hi @alextidd,

Yes, the development SCAN2 version (i.e., you'd have to clone this GitHub repo and install the r-scan2 package from GitHub rather than rely on the install from conda) does have new features for this that seem to work well. 

A manuscript was submitted to bioRxiv last week describing how to relax SCAN2 filters and then feed into the Sequoia algorithm to filter out artifacts that creep in when the filters are relaxed and to build phylogenies. Still waiting on bioRxiv to approve the submission.

The critical new capability is the `update.static.filter.params()` function, which allows the bulk-related calling parameters `max.bulk.alt`, `max.bulk.af` and `max.bulk.binom.prob` to be varied without rerunning the whole (long) SCAN2 pipeline.

I will try to remember to update this thread when the manuscript is posted.

**Comment by alextidd** [comment]
Awesome, thanks so much! I look forward to reading the manuscript :)

**Comment by jluquette** [comment]
Hi @alextidd - the preprint is now available on bioRxiv: https://www.biorxiv.org/content/10.1101/2025.10.31.685648v1

**Comment by alextidd** [comment]
Great read, thanks for sharing! Just to clarify, if I have already started running the conda version of SCAN2, should I stop it and then restart using the repo version? Or are the changes only downstream in the R package's processing of SCAN2 outputs?

**Comment by alextidd** [comment]
Hi @jluquette, I was just wondering, when will the 'develop' version of SCAN2 become the 'main' version? My institute is planning to set up SCAN2 as a module for people to use more easily, but IT was asking whether to hold of in case the repo will be changing to reflect the develop version anytime soon?
Thanks!
Alex

**Comment by jluquette** [comment]
Hi Alex,

Updating the SCAN2 packages is certainly on my to-do list. Changing the head on GitHub is easy, but rebuilding all of the conda packages, updating the (large) environment and the READMEs would take a good deal of effort.

How far into the future would be too long for your IT department?

**Comment by alextidd** [comment]
Hey @jluquette,
I have asked IT for their timeline and will get back to you when I hear. I have another question about the develop version (sorry for all the questions!)

When running the 'develop' version, do you run it inside of the scan2 conda environment, or are you running it in some other environment? When I run the conda version, it works fine...

```bash
# conda env
conda activate scan2

# initiate outdir
scan2 -d $out_dir init

Creating new analysis with default parameters in '/lustre/scratch125/casm/teams/team268/at31/projects/hashimoto_thyroiditis/out/resolveome/scan2/demo_develop'..
UNLOCKED SCAN2 analysis type=call_mutations (change with `config --analysis`), ID=0c17be08-07ba-486c-86aa-6737b85fb14d
           BAMs: 0 single cell(s), 0 bulk(s), 0 other
        Creator: at31
    Create date: 2026-02-05 09:47 GMT
Done.
Provide input files and set parameters via scan2 config.
Start the analysis with scan2 run.
```

However, when I run the repo version, it does not...

```bash
# conda env
conda activate scan2

# initate outdir
./repos/SCAN2/bin/scan2 -d $out_dir init

Traceback (most recent call last):
  File "/nfs/casm/team268im/at31/bin/repos/SCAN2/bin/scan2", line 16, in <module>
    import snakemake.cli
ModuleNotFoundError: No module named 'snakemake.cli'
```

It seems that the conda version of scan2 uses `snakemake.main()`, which is compatible with snakemake v7.x, whereas the repo version uses `snakemake.cli`, which requires snakemake v8+. The version of snakemake in the scan2 conda environment is 7.32.4.

I tried to update the snakemake version in the conda environment...

```bash
conda install -c conda-forge -c bioconda snakemake=8

Retrieving notices: / Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='conda.anaconda.org', port=443): Read timed out. (read timeout=5)")': /nodefaults/notices.json

done

Channels:
 - conda-forge
 - bioconda
 - nodefaults
 - dranew
 - soil
 - jluquette
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: - warning  libmamba Added empty dependency for problem type SOLVER_RULE_UPDATE
failed

LibMambaUnsatisfiableError: Encountered problems while solving:
  - nothing provides snakemake-minimal 8.11.2.* needed by snakemake-8.11.2-hdfd78af_0

Could not solve for environment specs
The following packages are incompatible
├─ pin-1 is installable and it requires
│  └─ python 3.10.* , which can be installed;
└─ snakemake 8**  is not installable because there are no viable options
   ├─ snakemake [8.0.0|8.0.1|...|8.9.0] would require
   │  └─ snakemake-minimal [8.0.0.* |8.0.1.* |...|8.9.0.* ], which requires
   │     └─ python >=3.11,<3.13 , which conflicts with any installable versions previously reported;
   └─ snakemake 8.11.2 would require
      └─ snakemake-minimal 8.11.2.* , which does not exist (perhaps a missing channel).
```

However, it looks like snakemake v8+, which is required for `snakemake.cli()`, is not compatible with the version of python in the scan2 environment (v3.10).

Have you experience this yourself and, if so, do you have a way around it?
Thanks so much again!!

**Comment by jluquette** [comment]
Yes, the development version of SCAN2 requires a much different environment, both for updated and new packages.

Here is the script I use to build the SCAN2 conda env. Some of the packages are not necessary for SCAN2 (I just use them routinely in my downstream analyses), but it shouldn't be too burdensome.

[install_BIG_scan2_env-3.sh](https://github.com/user-attachments/files/25134515/install_BIG_scan2_env-3.sh)
